src/modules/Dashboard/features/Applet/Schedule/ClearScheduledEventsPopup/ClearScheduledEventsPopup.tsx

```diff
@@ -2,23 +2,44 @@ import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
 
 import { Modal, SubmitBtnColor } from 'shared/components';
-import { StyledModalWrapper } from 'shared/styles/styledComponents';
+import { StyledModalWrapper } from 'shared/styles';
+import { useAsync } from 'shared/hooks';
+import { deleteScheduledEventsApi } from 'api';
+import { useAppDispatch } from 'redux/store';
+import { applets } from 'modules/Dashboard/state';
 
 import { ClearScheduledEventsPopupProps, Steps } from './ClearScheduledEventsPopup.types';
-import { getScreens } from './ClearScheduleEventsPopup.const';
+import { getScreens } from './ClearScheduleEventsPopup.utils';
 
 export const ClearScheduledEventsPopup = ({
   open,
   onClose,
   name,
   appletName,
+  appletId,
   isDefault = true,
 }: ClearScheduledEventsPopupProps) => {
   const { t } = useTranslation();
+  const dispatch = useAppDispatch();
   const [step, setStep] = useState<Steps>(0);
+  const { execute: deleteScheduledEvents } = useAsync(deleteScheduledEventsApi, () =>
+    dispatch(applets.thunk.getEvents({ appletId })),
+  );
+
+  const getNextStep = () =>
+    setStep((prevStep) => {
+      const newStep = prevStep + 1;
+
+      return newStep as Steps;
+    });
+  const onSubmit = async () => {
+    if (isDefault) {
+      await deleteScheduledEvents({ appletId });
+
+      return getNextStep();
```

Comment:
do y need   return getNextStep();? y have it bellow

Reply:
there will be another call to the api, if the schedule is not default